pkg/controller/nodes/node_exec_context.go

```diff
@@ -152,6 +160,17 @@ func (c *nodeExecutor) newNodeExecContextDefault(ctx context.Context, w v1alpha1
 		return nil
 	}
 
+	interrutible := w.IsInterruptible()
+	if n.IsInterruptible() != nil {
+		interrutible = *n.IsInterruptible()
+	}
+
+	// a node is not considered interruptible if the system failures have exceeded the configured threshold
+	if interrutible && s.GetSystemFailures() >= c.interruptibleFailureThreshold {
```

Comment:
so interruptions are clubbed with system failures is it?

Reply:
is it because there is no specific error that is observed during interruptions?

Reply:
Yes :(

Unfortunately, we can distinguish kubelet failures from node interruptions or manual node deletions.

Reply:
Apparently with a bit more work, we can: https://github.com/kube-aws/kube-spot-termination-notice-handler
I've not google much but either this or probably another project or we can just modify it to set a label on the pod instead or some other way to mark as being interrupted...

Reply:
We are actually planning to do this in V1 where we pass term signal to the containers as well because that is some work (cloudwatch listener -> ds -> easy support in SDK through a callback).
For V0, we want it to be a simple system error with some info in description.